core/src/main/java/nl/nn/adapterframework/pipes/Base64Pipe.java

```diff
@@ -134,18 +138,18 @@ public PipeRunResult doPipe(Message message, IPipeLineSession session) throws Pi
 			}
 			return target.getPipeRunResult();
 		} catch (Exception e) {
-			throw new PipeRunException(this, "cannot convert base64 "+getDirection()+" result", e);
+			throw new PipeRunException(this, "cannot convert base64 "+getDirectionEnum()+" result", e);
 		}
 	}
 
 	@SuppressWarnings("resource")
 	@Override
 	public MessageOutputStream provideOutputStream(IPipeLineSession session) throws StreamingException {
 		MessageOutputStream target = getTargetStream(session);
-		boolean directionEncode = "encode".equals(getDirection());//TRUE encode - FALSE decode
+		boolean directionEncode = getDirectionEnum()==Direction.ENCODE;//TRUE encode - FALSE decode
 		OutputStream targetStream;
-		if (!directionEncode && StringUtils.isNotEmpty(getCharset())) {
-			targetStream = new WriterOutputStream(target.asWriter(), getCharset());
+		if (getOutputTypeEnum()==OutputTypes.STRING || getOutputTypeEnum()==OutputTypes.STREAM && directionEncode || !directionEncode && StringUtils.isNotEmpty(getCharset())) {
+			targetStream = new WriterOutputStream(target.asWriter(), getCharset()!=null? getCharset() : StreamUtil.DEFAULT_INPUT_STREAM_ENCODING );
```

Comment:
Ik vind dat hij gewoon null mag doorgeven.. Dan doet hij zelf `StreamUtil.DEFAULT_INPUT_STREAM_ENCODING`

Reply:
Maar dat doet de WriterOutputStream niet

Reply:
Ah, die is niet van ons maar van Apache!